modules/lotamePanoramaIdSystem.js

```diff
@@ -200,24 +200,25 @@ export const lotamePanoramaIdSubmodule = {
       ajax(
         url,
         (response) => {
-          let responseObj = {};
+          let coreId;
           if (response) {
             try {
-              responseObj = JSON.parse(response);
+              let responseObj = JSON.parse(response);
               saveLotameCache(KEY_EXPIRY, responseObj.expiry_ts);
 
-              if (utils.isStr(responseObj.core_id)) {
-                saveLotameCache(
-                  KEY_ID,
-                  responseObj.core_id,
-                  responseObj.expiry_ts
-                );
-              } else {
-                clearLotameCache(KEY_ID);
-              }
-
               if (utils.isStr(responseObj.profile_id)) {
                 setProfileId(responseObj.profile_id);
+
+                if (utils.isStr(responseObj.core_id)) {
```

Comment:
hmmm - the assumption here is that we would always have a profile_id if we had a core_id. I could see a world where that wasn't true in the future where simply using attributes of the request was enough to generate the core_id but there was no corresponding profile_id. Might be worth getting Jeremy's opinion on that. How hard would it be to handle the existence or non-existence of either individually?

Reply:
Shouldn't wait until that is a thing? Today you can't have a core_id without a profile_id and the absence of a profile_id means we've opted out so we would need to send a different signal if core_id and no profile_id is a valid state

Reply:
ah forgot about the opted out part. I think the answer then in the future is always to return a value from the server for profile_id even if it is just a static representative value.